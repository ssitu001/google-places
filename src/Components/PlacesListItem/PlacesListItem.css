.list-group-item-alternate {
  background: #ececec;
}