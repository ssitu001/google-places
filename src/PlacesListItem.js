import React from 'react';

const PlacesListItem = ({place, index, displayInfoFromListItem}) => {

  return (
    <div className="list-group-item" onClick={() => displayInfoFromListItem(index)}>
      <h5 className="list-group-item-heading">
        {place.name}
      </h5>
      <p>{place.formatted_address}</p>
    </div>
  )
}

export default PlacesListItem;