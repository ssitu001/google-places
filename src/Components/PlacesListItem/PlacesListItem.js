import React from 'react';

import './PlacesListItem.css';

const PlacesListItem = ({place, index, displayInfoFromListItem}) => {

  return (
    <div 
      className={`list-group-item ${index % 2 === 0 ? 'list-group-item-alternate' : ''}`}
      onClick={() => displayInfoFromListItem(index)}>
      <h5 className="list-group-item-heading">
        {place.name}
      </h5>
      <p>{place.formatted_address}</p>
    </div>
  )
}

export default PlacesListItem;