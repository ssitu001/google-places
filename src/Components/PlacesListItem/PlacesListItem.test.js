import React from 'react';
import toJson from 'enzyme-to-json';
import PlacesListItem from './PlacesListItem';

const place = {
  name: 'Place1',
  formatted_address: '123 Fake St',
};

test('renders PlacesListItem component as expected', () => {
  const component = shallow(<PlacesListItem place={place}/>);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});

test('render PLaceListItem with alternate background', () => {
  const component = shallow(<PlacesListItem place={place} index={0}/>);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();  
});

test('render PLaceListItem with no background', () => {
  const component = shallow(<PlacesListItem place={place} index={1}/>);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();  
});