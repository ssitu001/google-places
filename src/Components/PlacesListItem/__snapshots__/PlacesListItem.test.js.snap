// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render PLaceListItem with alternate background 1`] = `
<div
  className="list-group-item list-group-item-alternate"
  onClick={[Function]}
>
  <h5
    className="list-group-item-heading"
  >
    Place1
  </h5>
  <p>
    123 Fake St
  </p>
</div>
`;

exports[`render PLaceListItem with no background 1`] = `
<div
  className="list-group-item "
  onClick={[Function]}
>
  <h5
    className="list-group-item-heading"
  >
    Place1
  </h5>
  <p>
    123 Fake St
  </p>
</div>
`;

exports[`renders PlacesListItem component as expected 1`] = `
<div
  className="list-group-item "
  onClick={[Function]}
>
  <h5
    className="list-group-item-heading"
  >
    Place1
  </h5>
  <p>
    123 Fake St
  </p>
</div>
`;
